src/views/RequiredSignoffs/ViewSignoff/index.jsx

```diff
@@ -61,46 +62,72 @@ const useStyles = makeStyles(theme => ({
 }));
 
 function ViewSignoff({ isNewSignoff, ...props }) {
-  const getEmptyRole = (id = 0) => ['', '', { isAdditionalRole: true, id }];
+  const getEmptyRole = (id = 0) => ({
+    name: '',
+    signoffs_required: null,
+    data_version: null,
+    sc: null,
+    metadata: {
+      isAdditionalRole: true,
+      id,
+    },
+  });
   const { product, channel } = props.match.params;
   const classes = useStyles();
   const [channelTextValue, setChannelTextValue] = useState(channel || '');
   const [productTextValue, setProductTextValue] = useState(product || '');
   const [type, setType] = useState(channel ? 'channel' : 'permission');
   const [roles, setRoles] = useState([]);
+  const [originalRoles, setOriginalRoles] = useState([]);
   const [additionalRoles, setAdditionalRoles] = useState(
     isNewSignoff ? [getEmptyRole()] : []
   );
-  const [sleepAction, sleep] = useAction(
-    timeout =>
-      new Promise((resolve, reject) =>
-        setTimeout(() => reject(new Error('Action not complete')), timeout)
-      )
-  );
   const [requiredSignoffs, getRS] = useAction(getRequiredSignoffs);
   const [products, fetchProducts] = useAction(getProducts);
   const [channels, fetchChannels] = useAction(getChannels);
+  const [saveAction, saveRS] = useAction(updateRequiredSignoffs);
   const isLoading =
     requiredSignoffs.loading || products.loading || channels.loading;
-  const error = requiredSignoffs.error || products.error || sleepAction.error;
+  const error = requiredSignoffs.error || products.error || saveAction.error;
   const handleTypeChange = ({ target: { value } }) => setType(value);
   const handleChannelChange = value => setChannelTextValue(value);
   const handleProductChange = value => setProductTextValue(value);
-  const handleRoleValueChange = (role, index) => ({ floatValue: value }) => {
-    const setRole = (entry, i) =>
-      i === index ? [entry[0], value, entry[2]] : entry;
+  const handleRoleValueChange = role => ({ floatValue: value }) => {
+    const setRole = entry => {
+      if (entry.name !== role.name) {
+        return entry;
+      }
+
+      const result = Object.assign(entry);
```

Comment:
I think the `no-param-reassign` rule is there to make sure the `arguments` object is not mutated. We could probably just do

```jsx
const setRole = entry => {
  const result = entry;
  
  // rest of logic
};
```

https://stackoverflow.com/a/42399879/6150432

I think there's another place where we can remove `Object.assign`.

Reply:
I'll give this a shot, thank you!